### PR TITLE
[Term Entry] Python:NumPy ndarray .tolist()

### LINKED ISSUE #4828: [Term Entry] Python:NumPy ndarray .tolist()
### Reason/inspiration (optional)

We would like a new entry on the `tolist` term under Numpy, `ndarray` concept. The entry should go in a new file under `docs/content/numpy/concepts/ndarray/terms/tolist/tolist.md`.

The entry should include:

- An introduction to the concept
- A ‘Syntax’ section that provides the syntax for the concept (if applicable)
- An ‘Example’ section that provides an example demonstrating the concept in use
- A ‘Codebyte Example’ section that provides a codebyte example demonstrating the concept in use

Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by jmargitic**
### Description

Hi!
I've added an entry for .tolist() for the ndarray content terms folder. It refers to issue: [Term Entry] Python:NumPy ndarray .tolist() #4828

### Issue Solved

Please use the following format(s) to link the issue numbers that your contribution addresses:
Closes #4828

### Type of Change

X Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

Hi! I believe I'm ready to submit my first pull request, however I have a few doubts with regards to the checklist.

With regards to point 1
_All writings are my own_

I've used resources from:
- https://numpy.org/doc/stable/reference/generated/numpy.ndarray.tolist.html
- ChatGPT (for editing the md content)

How can I properly cite this?

### CODE REVIEW

File: content/numpy/concepts/ndarray/terms/tolist/tolist.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+arr = np.uint32([1, 2])
+
+# Converting the NumPy array to a Python list using list()
+arr_list = list(arr)
+
+#Checking Output
+arr_list
+
+# Checking Output's type 
+type(arr_list[0])
```
Comment (mamtawardhani) [inline content/numpy/concepts/ndarray/terms/tolist/tolist.md:51]:
```suggestion
#Checking Output
print(arr_list)

# Checking Output's type 
print(type(arr_list[0]))
```

Comment (mamtawardhani) [reply content/numpy/concepts/ndarray/terms/tolist/tolist.md:51]: used the `print` statement for the output

File: content/numpy/concepts/ndarray/terms/tolist/tolist.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+# Converting the NumPy array to a Python list using .tolist()
+arr_tolist = arr.tolist()
+
+#Checking Output
+arr_tolist
+
+# Checking Output's type 
+type(arr_tolist[0])
```
Comment (mamtawardhani) [inline content/numpy/concepts/ndarray/terms/tolist/tolist.md:60]:
```suggestion
#Checking Output
print(arr_tolist)

# Checking Output's type 
print(type(arr_tolist[0]))
```

Comment (mamtawardhani) [reply content/numpy/concepts/ndarray/terms/tolist/tolist.md:60]: used the print statement for the output

File: content/numpy/concepts/ndarray/terms/tolist/tolist.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+arr_list = arr.tolist()
+
+print(arr_list)
+```
+
+This produces the following Output:
+
+``` python
+[[1, 2, 3], [4, 5, 6]]
+```
```
Comment (mamtawardhani) [inline content/numpy/concepts/ndarray/terms/tolist/tolist.md:97]: This code can be added inside the ##Codebyte Example block

File: content/numpy/concepts/ndarray/terms/tolist/tolist.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+<class 'numpy.uint32'>
+
+[1, 2]
+
+<class 'int'>
+```
+
+As we can see, although the resulting lists are similar, the outputs' type differ.
+
+### 2D Array
```
Comment (mamtawardhani) [inline content/numpy/concepts/ndarray/terms/tolist/tolist.md:77]:
```suggestion
## Codebyte Example
```

File: content/numpy/concepts/ndarray/terms/tolist/tolist.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+For a 2D array, `.tolist()` applies recursively:
```
Comment (mamtawardhani) [inline content/numpy/concepts/ndarray/terms/tolist/tolist.md:79]:
```suggestion
Run the following code to understand the working of the `.tolist()` method for converting a 2D NumPy array to a Python list:
```

File: content/numpy/concepts/ndarray/terms/tolist/tolist.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+``` python
```
Comment (mamtawardhani) [inline content/numpy/concepts/ndarray/terms/tolist/tolist.md:81]:
```suggestion
```codebyte/python
```

File: content/numpy/concepts/ndarray/terms/tolist/tolist.md (hunk 3 above)
Comment (mamtawardhani) [inline content/numpy/concepts/ndarray/terms/tolist/tolist.md:97]:
```suggestion
```

Comment (mamtawardhani) [reply content/numpy/concepts/ndarray/terms/tolist/tolist.md:97]: Codebyte blocks don't have output section

File: content/numpy/concepts/ndarray/terms/tolist/tolist.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+Return a copy of the array data as a (nested) Python list. Data items are converted to the nearest compatible built-in Python type via the **`item`** function.
+
+If `a.ndim` is 0, then since the depth of the nested list is 0, it will not be a list at all but a simple Python scalar.
+
+## Examples
+
+The following examples create a `ndarray` named `arr` and convert it into a nested Python list `arr_list` using `.tolist()`. We give examples of a 1D array and a 2D array.
+
+### 1D Array
+
```
Comment (mamtawardhani) [inline content/numpy/concepts/ndarray/terms/tolist/tolist.md:35]:
```suggestion
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @jmargitic, thank you contributing to Codecademy Docs! The entry is well written! 😃 

I've suggested a few changes, could you review and modify those at your earliest convenience? Feel free to discuss any concerns, thank you! 😄

**Comment by jmargitic** [comment]
Hi! 
This is ready to be reviewed again!
Thank you!

**Review by mamtawardhani — APPROVED** [review]
Thank you for making the changes and for your contribution @jmargitic! 😃 

The entry looks nice for a second round of review 🚀

**Review by Sriparno08 — APPROVED** [review]
Looks good, @jmargitic! 👍

**Pull request merged**